torch/_inductor/ir.py

```diff
@@ -2373,33 +2402,39 @@ def realize_input(cls, x):
 
     @classmethod
     def require_stride1(cls, x):
-        if len(x.get_stride()) == 0:
-            return x
-        for stride in x.get_stride():
-            if stride == 1:
+        if is_storage_and_layout(x):
+            if len(x.get_stride()) == 0:
                 return x
+            for stride in x.get_stride():
+                if stride == 1:
+                    return x
         return cls.copy_input(x)
 
     @classmethod
     def require_stride_order(cls, x, order):
         # require x to have the layout as strided_ordered as order
-        if isinstance(
-            x.get_layout(), FlexibleLayout
-        ) and is_stride_order_storage_and_layout(x, order):
-            # fix flexiblelayout to be FixedLayout with stride_order
-            as_storage_and_layout(
-                x, freeze=True, want_contiguous=False, stride_order=order
-            )
-            return x
-        elif isinstance(x.get_layout(), FixedLayout) and x.layout.is_stride_ordered(
-            order
-        ):
-            return x
+        if is_storage_and_layout(x):
```

Comment:
**is_storage_and_layout** need x is TensorBox, StorageBox or ReinterpretView, so if x is **InputBuffer** or **ComputedBuffer**, it always has a copy, I think it doesn't make sense.

Reply:
Good catch. I think the reason at the moment is that InputBuffer doesn't have a storage, but there's no reason it shouldn't (see: [TODO: input aliasing](https://github.com/pytorch/pytorch/blob/master/torch/_inductor/graph.py#L202).) 

However this particular problem (copy on the input) will get fixed by making the output here a FixedLayout in https://github.com/pytorch/pytorch/pull/88706 because we no longer call `apply_constraints`.